ROL: error building tests with Tpetra deprecated code off

## Bug Report
@trilinos/rol 

### Description
There's an error building ROL tests with Tpetra deprecated code off.
```
Trilinos/packages/rol/adapters/tpetra/src/function/ROL_TpetraBoundConstraint.hpp:374:46: error: no matching function for call to ‘Tpetra::MultiVector<double, int, long long int, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Serial> >::getLocalViewDevice()’
```
https://sems-cdash-son.sandia.gov/cdash/viewBuildError.php?buildid=29876

Also see https://github.com/sandialabs/Albany/issues/796

--- a/packages/rol/adapters/tpetra/src/function/ROL_TpetraBoundConstraint.hpp
+++ b/packages/rol/adapters/tpetra/src/function/ROL_TpetraBoundConstraint.hpp
@@ -132,14 +132,14 @@ namespace ROL {
         //----------------------------------------------------------------------
         //
         // Project x onto the bounds
-        template<class Real, class V>
+        template<class Real, class V, class CV>
         struct Project {
             typedef typename V::execution_space execution_space;
             V X_; // Optimization variable
-            V L_; // Lower bounds
-            V U_; // Upper bounds
+            CV L_; // Lower bounds
+            CV U_; // Upper bounds
 
-            Project(V& X, const V& L, const V& U) : X_(X), L_(L), U_(U) {}
+            Project(V& X, const CV& L, const CV& U) : X_(X), L_(L), U_(U) {}
 
             KOKKOS_INLINE_FUNCTION
             void operator() (const int i) const {
@@ -158,15 +158,15 @@ namespace ROL {
         //----------------------------------------------------------------------
         //
         // Set variables to zero if they correspond to the lower active set
-        template<class Real, class V>
+        template<class Real, class V, class CV>
         struct PruneLowerActive {
             typedef typename V::execution_space execution_space;
             V Y_; // Variable to be pruned
-            V X_; // Optimization variable
-            V L_; // Lower bounds
+            CV X_; // Optimization variable
+            CV L_; // Lower bounds
             Real eps_;
 
-            PruneLowerActive(V &Y, const V &X, const V &L,  Real eps) :
+            PruneLowerActive(V &Y, const CV &X, const CV &L,  Real eps) :
                 Y_(Y), X_(X), L_(L), eps_(eps) {}
 
             KOKKOS_INLINE_FUNCTION
@@ -184,15 +184,15 @@ namespace ROL {
         //----------------------------------------------------------------------
         //
         // Set variables to zero if they correspond to the upper active set
-        template<class Real, class V>
+        template<class Real, class V, class CV>
         struct PruneUpperActive {
             typedef typename V::execution_space execution_space;
             V Y_; // Variable to be pruned
-            V X_; // Optimization variable
-            V U_; // Upper bounds
+            CV X_; // Optimization variable
+            CV U_; // Upper bounds
             Real eps_;
 
-            PruneUpperActive(V &Y, const V &X, const V &U, Real eps) :
+            PruneUpperActive(V &Y, const CV &X, const CV &U, Real eps) :
                 Y_(Y), X_(X), U_(U), eps_(eps) {}
 
             KOKKOS_INLINE_FUNCTION
@@ -210,16 +210,16 @@ namespace ROL {
         //----------------------------------------------------------------------
         //
         // Set variables to zero if they correspond to the active set
-        template<class Real, class V>
+        template<class Real, class V, class CV>
         struct PruneActive {
             typedef typename V::execution_space execution_space;
             V Y_; // Variable to be pruned
-            V X_; // Optimization variable
-            V L_; // Lower bounds
-            V U_; // Upper bounds
+            CV X_; // Optimization variable
+            CV L_; // Lower bounds
+            CV U_; // Upper bounds
             Real eps_;
 
-            PruneActive(V &Y, const V &X, const V &L, const V &U, Real eps) :
+            PruneActive(V &Y, const CV &X, const CV &L, const CV &U, Real eps) :
                 Y_(Y), X_(X), L_(L), U_(U), eps_(eps) {}
 
             KOKKOS_INLINE_FUNCTION
@@ -240,17 +240,17 @@ namespace ROL {
         //----------------------------------------------------------------------
         //
         // Set variables to zero if they correspond to the lower active set and grad is positive
-        template<class Real, class V>
+        template<class Real, class V, class CV>
         struct GradPruneLowerActive {
             typedef typename V::execution_space execution_space;
             V Y_; // Variable to be pruned
-            V G_; // Gradient
-            V X_; // Optimization variable
-            V L_; // Lower bounds
+            CV G_; // Gradient
+            CV X_; // Optimization variable
+            CV L_; // Lower bounds
 
             Real xeps_, geps_;
 
-            GradPruneLowerActive(V &Y, const V &G, const V &X, const V &L,Real xeps, Real geps) :
+            GradPruneLowerActive(V &Y, const CV &G, const CV &X, const CV &L,Real xeps, Real geps) :
                 Y_(Y), G_(G), X_(X), L_(L), xeps_(xeps), geps_(geps) {}
 
             KOKKOS_INLINE_FUNCTION
@@ -268,17 +268,17 @@ namespace ROL {
         //----------------------------------------------------------------------
         //
         // Set variables to zero if they correspond to the upper active set and grad is negative
-        template<class Real, class V>
+        template<class Real, class V, class CV>
         struct GradPruneUpperActive {
             typedef typename V::execution_space execution_space;
             V Y_; // Variable to be pruned
-            V G_; // Gradient
-            V X_; // Optimization variable
-            V U_; // Upper bounds
+            CV G_; // Gradient
+            CV X_; // Optimization variable
+            CV U_; // Upper bounds
 
             Real xeps_, geps_;
 
-            GradPruneUpperActive(V &Y, const V &G, const V &X, const V &U, Real xeps, Real geps) :
+            GradPruneUpperActive(V &Y, const CV &G, const CV &X, const CV &U, Real xeps, Real geps) :
                 Y_(Y), G_(G), X_(X), U_(U), xeps_(xeps), geps_(geps) {}
 
             KOKKOS_INLINE_FUNCTION
@@ -296,17 +296,17 @@ namespace ROL {
         //----------------------------------------------------------------------
         //
         // Set variables to zero if they correspond to the active set
-        template<class Real, class V>
+        template<class Real, class V, class CV>
         struct GradPruneActive {
             typedef typename V::execution_space execution_space;
             V Y_; // Variable to be pruned
-            V G_; // Gradient
-            V X_; // Optimization variable
-            V L_; // Lower bounds
-            V U_; // Upper bounds
+            CV G_; // Gradient
+            CV X_; // Optimization variable
+            CV L_; // Lower bounds
+            CV U_; // Upper bounds
             Real xeps_, geps_;
 
-            GradPruneActive(V &Y, const V &G, const V &X, const V &L, const V &U, Real xeps, Real geps) :
+            GradPruneActive(V &Y, const CV &G, const CV &X, const CV &L, const CV &U, Real xeps, Real geps) :
                 Y_(Y), G_(G), X_(X), L_(L), U_(U), xeps_(xeps), geps_(geps) {}
 
             KOKKOS_INLINE_FUNCTION
@@ -335,20 +335,18 @@ namespace ROL {
 
 
         typedef Tpetra::MultiVector<Real,LO,GO,Node> MV;
+        typedef Tpetra::MultiVector<const Real,LO,GO,Node> CMV;
         typedef ROL::Ptr<MV> MVP;
-        typedef ROL::Ptr<const MV> CMVP;
         typedef TpetraMultiVector<Real,LO,GO,Node> TMV;
         typedef ROL::Ptr<TMV> TMVP;
         typedef typename MV::dual_view_type::t_dev ViewType;
+        typedef typename CMV::dual_view_type::t_dev ConstViewType;
 
         private:
             int gblDim_;
             int lclDim_;
-            MVP lp_;
-            MVP up_;
-
-            ViewType l_;           // Kokkos view of Lower bounds
-            ViewType u_;           // Kokkos view of Upper bounds
+            ConstViewType l_;           // Kokkos view of Lower bounds
+            ConstViewType u_;           // Kokkos view of Upper bounds
             Real min_diff_;
             Real scale_;
             ROL::Ptr<const Teuchos::Comm<int> > comm_;
@@ -366,14 +364,12 @@ namespace ROL {
             TpetraBoundConstraint(MVP lp, MVP up, Real scale = 1.0) :
                 gblDim_(lp->getGlobalLength()),
                 lclDim_(lp->getLocalLength()),
-                lp_(lp),
-                up_(up),
-                l_(lp->getLocalViewDevice()),
-                u_(up->getLocalViewDevice()),
+                l_(lp->getLocalViewDevice(Tpetra::Access::ReadOnly)),
+                u_(up->getLocalViewDevice(Tpetra::Access::ReadOnly)),
                 scale_(scale),
                 comm_(lp->getMap()->getComm()) {
 
-                KokkosStructs::MinGap<Real,ViewType> findmin(l_,u_);
+                KokkosStructs::MinGap<Real,ConstViewType> findmin(l_,u_);
                 Real lclMinGap = 0;
 
                 // Reduce for this MPI process
@@ -393,9 +389,9 @@ namespace ROL {
 
                 int lclFeasible = 1;
 
-                ViewType x_lcl = xp->getLocalViewDevice();
+                ConstViewType x_lcl = xp->getLocalViewDevice(Tpetra::Access::ReadOnly);
 
-                KokkosStructs::Feasible<Real,ViewType> check(x_lcl, l_, u_);
+                KokkosStructs::Feasible<Real,ConstViewType> check(x_lcl, l_, u_);
 
                 Kokkos::parallel_reduce(lclDim_,check,lclFeasible);
 
@@ -410,9 +406,9 @@ namespace ROL {
 
                 auto xp = getVector(x);
 
-                ViewType x_lcl = xp->getLocalViewDevice();
+                ViewType x_lcl = xp->getLocalViewDevice(Tpetra::Access::ReadWrite);
 
-                KokkosStructs::Project<Real,ViewType> proj(x_lcl,l_,u_);
+                KokkosStructs::Project<Real,ViewType,ConstViewType> proj(x_lcl,l_,u_);
 
                 Kokkos::parallel_for(lclDim_,proj);
             }
@@ -424,10 +420,10 @@ namespace ROL {
 
                 Real epsn = std::min(scale_*eps,this->min_diff_);
 
-                ViewType x_lcl = xp->getLocalViewDevice();
-                ViewType v_lcl = vp->getLocalViewDevice();
+                ConstViewType x_lcl = xp->getLocalViewDevice(Tpetra::Access::ReadOnly);
+                ViewType v_lcl = vp->getLocalViewDevice(Tpetra::Access::ReadWrite);
 
-                KokkosStructs::PruneLowerActive<Real,ViewType> prune(v_lcl,x_lcl,l_,epsn);
+                KokkosStructs::PruneLowerActive<Real,ViewType,ConstViewType> prune(v_lcl,x_lcl,l_,epsn);
 
                 Kokkos::parallel_for(lclDim_,prune);
             }
@@ -438,10 +434,10 @@ namespace ROL {
 
                 Real epsn = std::min(scale_*eps,this->min_diff_);
 
-                ViewType x_lcl = xp->getLocalViewDevice();
-                ViewType v_lcl = vp->getLocalViewDevice();
+                ConstViewType x_lcl = xp->getLocalViewDevice(Tpetra::Access::ReadOnly);
+                ViewType v_lcl = vp->getLocalViewDevice(Tpetra::Access::ReadWrite);
 
-                KokkosStructs::PruneUpperActive<Real,ViewType> prune(v_lcl,x_lcl,u_,epsn);
+                KokkosStructs::PruneUpperActive<Real,ViewType,ConstViewType> prune(v_lcl,x_lcl,u_,epsn);
 
                 Kokkos::parallel_for(lclDim_,prune);
             }
@@ -452,10 +448,10 @@ namespace ROL {
 
                 Real epsn = std::min(scale_*eps,this->min_diff_);
 
-                ViewType x_lcl = xp->getLocalViewDevice();
-                ViewType v_lcl = vp->getLocalViewDevice();
+                ConstViewType x_lcl = xp->getLocalViewDevice(Tpetra::Access::ReadOnly);
+                ViewType v_lcl = vp->getLocalViewDevice(Tpetra::Access::ReadWrite);
 
-                KokkosStructs::PruneActive<Real,ViewType> prune(v_lcl,x_lcl,l_,u_,epsn);
+                KokkosStructs::PruneActive<Real,ViewType,ConstViewType> prune(v_lcl,x_lcl,l_,u_,epsn);
 
                 Kokkos::parallel_for(lclDim_,prune);
             }
@@ -466,11 +462,11 @@ namespace ROL {
                 auto vp = getVector(v);
                Real epsn = std::min(scale_*xeps,this->min_diff_);
 
-                ViewType x_lcl = xp->getLocalViewDevice();
-                ViewType g_lcl = gp->getLocalViewDevice();
-                ViewType v_lcl = vp->getLocalViewDevice();
+                ConstViewType x_lcl = xp->getLocalViewDevice(Tpetra::Access::ReadOnly);
+                ConstViewType g_lcl = gp->getLocalViewDevice(Tpetra::Access::ReadOnly);
+                ViewType v_lcl = vp->getLocalViewDevice(Tpetra::Access::ReadWrite);
 
-                KokkosStructs::GradPruneLowerActive<Real,ViewType> prune(v_lcl,g_lcl,x_lcl,l_,epsn,geps);
+                KokkosStructs::GradPruneLowerActive<Real,ViewType,ConstViewType> prune(v_lcl,g_lcl,x_lcl,l_,epsn,geps);
 
                 Kokkos::parallel_for(lclDim_,prune);
             }
@@ -481,11 +477,11 @@ namespace ROL {
                 auto vp = getVector(v);
                 Real epsn = std::min(scale_*xeps,this->min_diff_);
 
-                ViewType x_lcl = xp->getLocalViewDevice();
-                ViewType g_lcl = gp->getLocalViewDevice();
-                ViewType v_lcl = vp->getLocalViewDevice();
+                ConstViewType x_lcl = xp->getLocalViewDevice(Tpetra::Access::ReadOnly);
+                ConstViewType g_lcl = gp->getLocalViewDevice(Tpetra::Access::ReadOnly);
+                ViewType v_lcl = vp->getLocalViewDevice(Tpetra::Access::ReadWrite);
 
-                KokkosStructs::GradPruneUpperActive<Real,ViewType> prune(v_lcl,g_lcl,x_lcl,u_,epsn,geps);
+                KokkosStructs::GradPruneUpperActive<Real,ViewType,ConstViewType> prune(v_lcl,g_lcl,x_lcl,u_,epsn,geps);
 
                 Kokkos::parallel_for(lclDim_,prune);
             }
@@ -496,11 +492,11 @@ namespace ROL {
                 auto vp = getVector(v);
                 Real epsn = std::min(scale_*xeps,this->min_diff_);
 
-                ViewType x_lcl = xp->getLocalViewDevice();
-                ViewType g_lcl = gp->getLocalViewDevice();
-                ViewType v_lcl = vp->getLocalViewDevice();
+                ConstViewType x_lcl = xp->getLocalViewDevice(Tpetra::Access::ReadOnly);
+                ConstViewType g_lcl = gp->getLocalViewDevice(Tpetra::Access::ReadOnly);
+                ViewType v_lcl = vp->getLocalViewDevice(Tpetra::Access::ReadWrite);
 
-                KokkosStructs::GradPruneActive<Real,ViewType> prune(v_lcl,g_lcl,x_lcl,l_,u_,epsn,geps);
+                KokkosStructs::GradPruneActive<Real,ViewType,ConstViewType> prune(v_lcl,g_lcl,x_lcl,l_,u_,epsn,geps);
 
                 Kokkos::parallel_for(lclDim_,prune);
             }

--- a/packages/rol/adapters/tpetra/src/vector/ROL_TpetraMultiVector.hpp
+++ b/packages/rol/adapters/tpetra/src/vector/ROL_TpetraMultiVector.hpp
@@ -92,12 +92,13 @@ namespace TPMultiVector {
             class Node=Tpetra::Map<>::node_type >
    struct binaryFunc {
     typedef typename Tpetra::MultiVector<Real,LO,GO,Node>::dual_view_type::t_dev ViewType;
+    typedef typename Tpetra::MultiVector<const Real,LO,GO,Node>::dual_view_type::t_dev ConstViewType;
     typedef typename ViewType::execution_space execution_space;
     ViewType X_;
-    ViewType Y_;
+    ConstViewType Y_;
     const Elementwise::BinaryFunction<Real>* const f_;
 
-    binaryFunc(ViewType& X, const ViewType& Y, const Elementwise::BinaryFunction<Real>* const f)
+    binaryFunc(ViewType& X, const ConstViewType& Y, const Elementwise::BinaryFunction<Real>* const f)
       : X_(X), Y_(Y), f_(f) {}
 
     KOKKOS_INLINE_FUNCTION
@@ -115,12 +116,12 @@ namespace TPMultiVector {
             class GO=Tpetra::Map<>::global_ordinal_type,
             class Node=Tpetra::Map<>::node_type >
   struct reduceFunc {
-    typedef typename Tpetra::MultiVector<Real,LO,GO,Node>::dual_view_type::t_dev ViewType;
-    typedef typename ViewType::execution_space execution_space;
-    ViewType X_;
+    typedef typename Tpetra::MultiVector<const Real,LO,GO,Node>::dual_view_type::t_dev ConstViewType;
+    typedef typename ConstViewType::execution_space execution_space;
+    ConstViewType X_;
     const Elementwise::ReductionOp<Real>* const r_;
 
-    reduceFunc(const ViewType& X, const Elementwise::ReductionOp<Real>* const r)
+    reduceFunc(const ConstViewType& X, const Elementwise::ReductionOp<Real>* const r)
       : X_(X), r_(r) {}
 
     KOKKOS_INLINE_FUNCTION
@@ -287,10 +288,10 @@ public:
 /*****************************************************************************/
 private:
   typedef typename Tpetra::MultiVector<Real,LO,GO,Node>::dual_view_type::t_dev ViewType;
-
+  typedef typename Tpetra::MultiVector<const Real,LO,GO,Node>::dual_view_type::t_dev ConstViewType;
 public:
   void applyUnary( const Elementwise::UnaryFunction<Real> &f ) {
-    ViewType v_lcl =  tpetra_vec_->getLocalViewDevice();
+    ViewType v_lcl =  tpetra_vec_->getLocalViewDevice(Tpetra::Access::ReadWrite);
 
     int lclDim = tpetra_vec_->getLocalLength();
     TPMultiVector::unaryFunc<Real,LO,GO,Node> func(v_lcl, &f);
@@ -307,8 +308,8 @@ public:
    const TpetraMultiVector &ex = dynamic_cast<const TpetraMultiVector&>(x);
    Ptr<const Tpetra::MultiVector<Real,LO,GO,Node> > xp = ex.getVector();
 
-    ViewType v_lcl = tpetra_vec_->getLocalViewDevice();
-    ViewType x_lcl = xp->getLocalViewDevice();
+    ViewType v_lcl = tpetra_vec_->getLocalViewDevice(Tpetra::Access::ReadWrite);
+    ConstViewType x_lcl = xp->getLocalViewDevice(Tpetra::Access::ReadOnly);
 
     int lclDim = tpetra_vec_->getLocalLength();
 
@@ -319,7 +320,7 @@ public:
   }
 
   Real reduce( const Elementwise::ReductionOp<Real> &r ) const {
-    ViewType v_lcl = tpetra_vec_->getLocalViewDevice();
+    ConstViewType v_lcl = tpetra_vec_->getLocalViewDevice(Tpetra::Access::ReadOnly);
 
     int lclDim = tpetra_vec_->getLocalLength();
     TPMultiVector::reduceFunc<Real,LO,GO,Node> func(v_lcl, &r);

--- a/packages/rol/adapters/tpetra/test/vector/test_01.cpp
+++ b/packages/rol/adapters/tpetra/test/vector/test_01.cpp
@@ -191,7 +191,7 @@ int main(int argc, char *argv[]) {
         MVP hv_ptr    = ROL::makePtr<MV>(map,1,true);
         MVP ihhv_ptr  = ROL::makePtr<MV>(map,1,true);
  
-        int numElem = map->getNodeNumElements();
+        int numElem = map->getLocalNumElements();
          
         for( LO lclRow = 0; lclRow < static_cast<LO> (numElem); ++lclRow) {
             const GO gblRow = map->getGlobalElement(lclRow);

--- a/packages/rol/example/PDE-OPT/0ld/TOOLS/PDE_FEM.hpp
+++ b/packages/rol/example/PDE-OPT/0ld/TOOLS/PDE_FEM.hpp
@@ -278,13 +278,13 @@ public:
 
     // Build maps.
     myOverlapMap_ = ROL::makePtr<Tpetra::Map<>>(Teuchos::OrdinalTraits<Tpetra::global_size_t>::invalid(),                                            myGlobIds_, 0, commPtr_);
-    //std::cout << std::endl << myOverlapMap_->getNodeElementList()<<std::endl;
+    //std::cout << std::endl << myOverlapMap_->getLocalElementList()<<std::endl;
     /** One can also use the non-member function:
         myOverlapMap_ = Tpetra::createNonContigMap<int,int>(myGlobIds_, commPtr_);
         to build the overlap map.
     **/
     myUniqueMap_ = Tpetra::createOneToOne(myOverlapMap_);
-    //std::cout << std::endl << myUniqueMap_->getNodeElementList() << std::endl;
+    //std::cout << std::endl << myUniqueMap_->getLocalElementList() << std::endl;
     myCellMap_ = ROL::makePtr<Tpetra::Map<>>(Teuchos::OrdinalTraits<Tpetra::global_size_t>::invalid(),
                               this->myCellIds_, 0, this->commPtr_);
   }
@@ -387,7 +387,7 @@ public:
     /****************************************/
 
     Teuchos::ArrayRCP<const GO> cellDofsArrayRCP = cellDofs_.getData();
-    Teuchos::Array<size_t> graphEntriesPerRow(myUniqueMap_->getNodeNumElements());
+    Teuchos::Array<size_t> graphEntriesPerRow(myUniqueMap_->getLocalNumElements());
     for (GO i=0; i<numCells_; ++i) {
       for (int j=0; j<numLocalDofs_; ++j) {
         GO gid = cellDofs_(myCellIds_[i],j);

--- a/packages/rol/example/PDE-OPT/0ld/adv-diff-react/data.hpp
+++ b/packages/rol/example/PDE-OPT/0ld/adv-diff-react/data.hpp
@@ -258,13 +258,13 @@ public:
     // Build maps.
     myOverlapMap_ = ROL::makePtr<Tpetra::Map<>>(Teuchos::OrdinalTraits<Tpetra::global_size_t>::invalid(),
                                                    myGlobIds_, 0, comm);
-    //std::cout << std::endl << myOverlapMap_->getNodeElementList();
+    //std::cout << std::endl << myOverlapMap_->getLocalElementList();
     /** One can also use the non-member function:
           myOverlapMap_ = Tpetra::createNonContigMap<int,int>(myGlobIds_, comm);
         to build the overlap map.
     **/
     myUniqueMap_ = Tpetra::createOneToOne(myOverlapMap_);
-    //std::cout << std::endl << myUniqueMap_->getNodeElementList() << std::endl;
+    //std::cout << std::endl << myUniqueMap_->getLocalElementList() << std::endl;
 
     /****************************************************/
     /****************************************************/
@@ -433,7 +433,7 @@ public:
     // Assemble graph.
 
     Teuchos::ArrayRCP<const GO> cellDofsArrayRCP = cellDofs.getData();
-    Teuchos::Array<size_t> graphEntriesPerRow(myUniqueMap_->getNodeNumElements());
+    Teuchos::Array<size_t> graphEntriesPerRow(myUniqueMap_->getLocalNumElements());
     for (GO i=0; i<numCells_; ++i) {
       for (int j=0; j<numLocalDofs; ++j) {
         GO gid = cellDofs(myCellIds_[i],j);

--- a/packages/rol/example/PDE-OPT/0ld/elasticitySIMP_topologyOptimization/filter.hpp
+++ b/packages/rol/example/PDE-OPT/0ld/elasticitySIMP_topologyOptimization/filter.hpp
@@ -242,13 +242,13 @@ public:
     // Build maps.
     myOverlapMap_ = ROL::makePtr<Tpetra::Map<>>(Teuchos::OrdinalTraits<Tpetra::global_size_t>::invalid(),
                                                    myGlobIds_, 0, comm);
-    //std::cout << std::endl << myOverlapMap_->getNodeElementList();
+    //std::cout << std::endl << myOverlapMap_->getLocalElementList();
     /** One can also use the non-member function:
           myOverlapMap_ = Tpetra::createNonContigMap<int,int>(myGlobIds_, comm);
         to build the overlap map.
     **/
     myUniqueMap_ = Tpetra::createOneToOne(myOverlapMap_);
-    //std::cout << std::endl << myUniqueMap_->getNodeElementList() << std::endl;
+    //std::cout << std::endl << myUniqueMap_->getLocalElementList() << std::endl;
 
     myBColumnMap_ = ROL::makePtr<Tpetra::Map<>>(Teuchos::OrdinalTraits<Tpetra::global_size_t>::invalid(),
                                  myCellIds_, 0, comm);

--- a/packages/rol/example/PDE-OPT/0ld/poisson/data.hpp
+++ b/packages/rol/example/PDE-OPT/0ld/poisson/data.hpp
@@ -244,13 +244,13 @@ public:
     // Build maps.
     myOverlapMap_ = ROL::makePtr<Tpetra::Map<>>(Teuchos::OrdinalTraits<Tpetra::global_size_t>::invalid(),
                                                    myGlobIds_, 0, comm);
-    //std::cout << std::endl << myOverlapMap_->getNodeElementList();
+    //std::cout << std::endl << myOverlapMap_->getLocalElementList();
     /** One can also use the non-member function:
           myOverlapMap_ = Tpetra::createNonContigMap<int,int>(myGlobIds_, comm);
         to build the overlap map.
     **/
     myUniqueMap_ = Tpetra::createOneToOne(myOverlapMap_);
-    //std::cout << std::endl << myUniqueMap_->getNodeElementList() << std::endl;
+    //std::cout << std::endl << myUniqueMap_->getLocalElementList() << std::endl;
 
     /****************************************************/
     /****************************************************/
@@ -383,7 +383,7 @@ public:
 
     // Assemble graph.
     Teuchos::ArrayRCP<const GO> cellDofsArrayRCP = cellDofs.getData();
-    Teuchos::Array<size_t> graphEntriesPerRow(myUniqueMap_->getNodeNumElements());
+    Teuchos::Array<size_t> graphEntriesPerRow(myUniqueMap_->getLocalNumElements());
     for (GO i=0; i<numCells_; ++i) {
       for (int j=0; j<numLocalDofs; ++j) {
         GO gid = cellDofs(myCellIds_[i],j);

--- a/packages/rol/example/PDE-OPT/0ld/stefan-boltzmann/data.hpp
+++ b/packages/rol/example/PDE-OPT/0ld/stefan-boltzmann/data.hpp
@@ -251,13 +251,13 @@ public:
     // Build maps.
     myOverlapMap_ = ROL::makePtr<Tpetra::Map<>>(Teuchos::OrdinalTraits<Tpetra::global_size_t>::invalid(),
                                                    myGlobIds_, 0, comm);
-    //std::cout << std::endl << myOverlapMap_->getNodeElementList();
+    //std::cout << std::endl << myOverlapMap_->getLocalElementList();
     /** One can also use the non-member function:
           myOverlapMap_ = Tpetra::createNonContigMap<int,int>(myGlobIds_, comm);
         to build the overlap map.
     **/
     myUniqueMap_ = Tpetra::createOneToOne(myOverlapMap_);
-    //std::cout << std::endl << myUniqueMap_->getNodeElementList() << std::endl;
+    //std::cout << std::endl << myUniqueMap_->getLocalElementList() << std::endl;
 
     /****************************************************/
     /****************************************************/
@@ -405,7 +405,7 @@ public:
 
     // Assemble graph.
     Teuchos::ArrayRCP<const GO> cellDofsArrayRCP = cellDofs.getData();
-    Teuchos::Array<size_t> graphEntriesPerRow(myUniqueMap_->getNodeNumElements());
+    Teuchos::Array<size_t> graphEntriesPerRow(myUniqueMap_->getLocalNumElements());
     for (GO i=0; i<numCells_; ++i) {
       for (int j=0; j<numLocalDofs; ++j) {
         GO gid = cellDofs(myCellIds_[i],j);

--- a/packages/rol/example/PDE-OPT/0ld/stoch-adv-diff/data.hpp
+++ b/packages/rol/example/PDE-OPT/0ld/stoch-adv-diff/data.hpp
@@ -263,13 +263,13 @@ public:
     // Build maps.
     myOverlapMap_ = ROL::makePtr<Tpetra::Map<>>(Teuchos::OrdinalTraits<Tpetra::global_size_t>::invalid(),
                                                    myGlobIds_, 0, comm);
-    //std::cout << std::endl << myOverlapMap_->getNodeElementList();
+    //std::cout << std::endl << myOverlapMap_->getLocalElementList();
     /** One can also use the non-member function:
           myOverlapMap_ = Tpetra::createNonContigMap<int,int>(myGlobIds_, comm);
         to build the overlap map.
     **/
     myUniqueMap_ = Tpetra::createOneToOne(myOverlapMap_);
-    //std::cout << std::endl << myUniqueMap_->getNodeElementList() << std::endl;
+    //std::cout << std::endl << myUniqueMap_->getLocalElementList() << std::endl;
 
     /****************************************************/
     /****************************************************/
@@ -480,7 +480,7 @@ public:
 
     // Assemble graph.
     Teuchos::ArrayRCP<const GO> cellDofsArrayRCP = cellDofs.getData();
-    Teuchos::Array<size_t> graphEntriesPerRow(myUniqueMap_->getNodeNumElements());
+    Teuchos::Array<size_t> graphEntriesPerRow(myUniqueMap_->getLocalNumElements());
     for (GO i=0; i<numCells_; ++i) {
       for (int j=0; j<numLocalDofs; ++j) {
         GO gid = cellDofs(myCellIds_[i],j);

--- a/packages/rol/example/PDE-OPT/TOOLS/sysbuilder.hpp
+++ b/packages/rol/example/PDE-OPT/TOOLS/sysbuilder.hpp
@@ -107,9 +107,9 @@ public:
     ROL::Ptr<const Tpetra::Map<> > mapRes = A->getRangeMap();
 
     // Get individual indices.
-    Teuchos::ArrayView<const GO> idxSim = mapSim->getNodeElementList(); 
-    Teuchos::ArrayView<const GO> idxOpt = mapOpt->getNodeElementList(); 
-    Teuchos::ArrayView<const GO> idxRes = mapRes->getNodeElementList();
+    Teuchos::ArrayView<const GO> idxSim = mapSim->getLocalElementList();
+    Teuchos::ArrayView<const GO> idxOpt = mapOpt->getLocalElementList();
+    Teuchos::ArrayView<const GO> idxRes = mapRes->getLocalElementList();
 
     // Get communicator.
     ROL::Ptr<const Teuchos::Comm<int> > comm = mapSim->getComm();
@@ -347,9 +347,9 @@ public:
     ROL::Ptr<const Tpetra::Map<> > mapRes = vecRes->getMap();
 
     // Get individual indices.
-    Teuchos::ArrayView<const GO> idxSim = mapSim->getNodeElementList(); 
-    Teuchos::ArrayView<const GO> idxOpt = mapOpt->getNodeElementList(); 
-    Teuchos::ArrayView<const GO> idxRes = mapRes->getNodeElementList();
+    Teuchos::ArrayView<const GO> idxSim = mapSim->getLocalElementList();
+    Teuchos::ArrayView<const GO> idxOpt = mapOpt->getLocalElementList();
+    Teuchos::ArrayView<const GO> idxRes = mapRes->getLocalElementList();
 
     // Get communicator.
     ROL::Ptr<const Teuchos::Comm<int> > comm = mapSim->getComm();

--- a/packages/rol/src/utils/ROL_ScalarMinimizationTest01.hpp
+++ b/packages/rol/src/utils/ROL_ScalarMinimizationTest01.hpp
@@ -128,7 +128,7 @@ public:
       A = (Real)((i+1)*(i+1));
       B = (Real)((i+2)*(i+2));
       ScalarMinimizationTest<Real>::run(fx,x,nfval,ngrad,*f,A,B);
-      flag *= std::abs(x-xvector_[i]) < 10.0*std::sqrt(ROL_EPSILON<Real>())*xvector_[i];
+      flag &= std::abs(x-xvector_[i]) < 10.0*std::sqrt(ROL_EPSILON<Real>())*xvector_[i];
 
       stream << "  ";
       stream << std::setw(10) << std::left << (int)A;

--- a/packages/rol/test/elementwise/test_02.cpp
+++ b/packages/rol/test/elementwise/test_02.cpp
@@ -135,7 +135,7 @@ int main(int argc, char *argv[]) {
     V z(z_ptr);
     V z_thresh(z_thresh_ptr);
 
-    LO numElements = static_cast<LO>( map->getNodeNumElements() );
+    LO numElements = static_cast<LO>( map->getLocalNumElements() );
  
     for( LO lclRow = 0; lclRow < numElements; ++lclRow ) {
       const GO gblRow = map->getGlobalElement(lclRow);

